package com.eViewer.HTML5.Automation.Framework.Common;

import org.testng.Assert;

public class CustomAssertion {

	private static String failureMessage = "";

	public static <T> boolean assertEquals(T actual, T expected, String message) {
		boolean check = true;
		try {
			Assert.assertEquals(actual, expected, message);
		} catch (AssertionError e) {
			failureMessage = e.getMessage();
			CustomLogger.logError("assertEquals: check failed, " + failureMessage);
			check = false;
		}
		return check;
	}

	private static <T> boolean assertEquals(T actual, T expected) {
		boolean check = true;
		try {
			Assert.assertEquals(actual, expected);
		} catch (AssertionError e) {
			CustomLogger.logError("assertEquals: check failed, " + e.getMessage());
			check = false;
		}
		return check;
	}

	public static <T> boolean assertNotEquals(T actual, T expected, String message) {
		boolean check = true;
		try {
			Assert.assertNotEquals(actual, expected,
					message + " expected [" + expected + "] while found is [" + actual + "]");
		} catch (AssertionError e) {
			failureMessage = e.getMessage();
			CustomLogger.logError("assertNotEquals: check failed, " + failureMessage);
			check = false;
		}
		return check;
	}

	public static boolean assertTrue(boolean condition, String message) {
		boolean check = true;
		try {
			Assert.assertTrue(condition, message);
		} catch (AssertionError e) {
			failureMessage = e.getMessage();
			CustomLogger.logError("assertTrue: check failed, " + failureMessage);
			check = false;
		}
		return check;
	}

	public static boolean assertFalse(boolean condition, String message) {
		boolean check = true;
		try {
			Assert.assertFalse(condition, message);
		} catch (AssertionError e) {
			failureMessage = e.getMessage();
			CustomLogger.logError("assertFalse: check failed, " + failureMessage);
			check = false;
		}
		return check;
	}

	public static boolean assertStringContains(String actualString, String expectedToContain) {
		boolean check = true;
		try {
			if (!actualString.contains(expectedToContain)) {
				throw new AssertionError(
						"expected [" + expectedToContain + "] to be present in [" + actualString + "]");
			}
		} catch (AssertionError e) {
			failureMessage = e.getMessage();
			CustomLogger.logError("assertStringContains: check failed, " + failureMessage);
			check = false;
		}
		return check;
	}

	public static void assertTestPass(boolean actual, boolean expected) {
		if (!assertEquals(actual, expected)) {
			throw new AssertionError(
					"assertTestPass: final check for test case failed. Failure reason: " + failureMessage);
		}
	}

}
